Silicon/CommonSocPkg/Library/HeciLib/HeciLib.c

```diff
@@ -222,6 +222,49 @@ HeciGetFwCapsSkuMsg (
   return EFI_SUCCESS;
 }
 
+/**
+  Check for Manufacturing Mode
+
+  @param[out]  MeManuMode         Manufacturing mode/production mode.
+
+  @retval EFI_UNSUPPORTED         Current CSME mode doesn't support this function
+  @retval EFI_SUCCESS             Command succeeded
+
+**/
+EFI_STATUS
+EFIAPI
+HeciGetManufactureMode (
+  OUT UINT16 *MeManuMode
+  )
+{
+  UINTN                    HeciMmPciAddress;
+  HECI_FW_STS6_REGISTER    MeFirmwareStatus;
+
+  if (MeManuMode == NULL) {
+    return RETURN_INVALID_PARAMETER;
+  }
+
+  HeciMmPciAddress = MeGetHeciMmPciAddress (0, 0);
+  if (MmioRead16 (HeciMmPciAddress + PCI_VENDOR_ID_OFFSET) == 0xFFFF) {
+    return EFI_NOT_FOUND;
+  }
+
+  MeFirmwareStatus.ul = MmioRead32 (HeciMmPciAddress + R_ME_HFS_6);

```

Comment:
It looks some silicon does not have HFS_6 register. Can we get this information for MeChipsetLib? If a silicon does not provide a valid register info, we can just return UNSUPPORTED here.